# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/vm/PortableBaselineInterpret.cpp
# Commit: 364d4e24bc13
# Full Hash: 364d4e24bc13eafbb893adf6381a6cbb862d405f
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2024-07-29 20:43:57
# Regressor Bug: 1909224
# File Overlap Count: 2
# Description:
#   Bug 1909224 part 3 - Remove checks for FailInstanceReg on JIT => Wasm call path. r=rhunt
#   
#   Because we now unwind JS JIT frames after unwinding Wasm frames, we no longer
#   return to the JIT => Wasm entry code from the exception handler.
#   
# ==============================================================================

diff -r 6eb32b942a04 -r 364d4e24bc13 js/src/vm/PortableBaselineInterpret.cpp
--- a/js/src/vm/PortableBaselineInterpret.cpp	Mon Jul 29 13:14:47 2024 +0000
+++ b/js/src/vm/PortableBaselineInterpret.cpp	Mon Jul 29 13:14:48 2024 +0000
@@ -6128,8 +6128,10 @@
       case ExceptionResumeKind::Bailout:
         MOZ_CRASH(
             "Unexpected Bailout exception-resume kind in Portable Baseline");
-      case ExceptionResumeKind::Wasm:
-        MOZ_CRASH("Unexpected Wasm exception-resume kind in Portable Baseline");
+      case ExceptionResumeKind::WasmInterpEntry:
+        MOZ_CRASH(
+            "Unexpected WasmInterpEntry exception-resume kind in Portable "
+            "Baseline");
       case ExceptionResumeKind::WasmCatch:
         MOZ_CRASH(
             "Unexpected WasmCatch exception-resume kind in Portable "